firebase-vscode/src/core/config.ts

```diff
@@ -39,10 +46,24 @@ export async function updateFirebaseRCProject(
   rc.save();
 }
 
+
 function notifyFirebaseConfig(broker: ExtensionBrokerImpl) {
   broker.send("notifyFirebaseConfig", {
-    firebaseJson: firebaseConfig.value?.data,
-    firebaseRC: firebaseRC.value?.data,
+    firebaseJson: firebaseConfig.value?.switchCase<
```

Comment:
similar thoughts here. Is there enough value prop to warrant this level of code to return a value. Is it not sufficient to error out on read and maintain firebaseConfig as value | undefined.

Reply:
`value | undefined` is not enough.
We have 3 separate states:
- no value because configs are still lloading
- configs are loaded, but the config file is missing
- configs are loaded, and the file is malformed


The current type differentiates all those states.

`undefined` means "Configs" are loading
`ResultValue(undefined)` means configs loading has completed, and the file doesn't exist
`ResultError(...)` means there's an error in the config file.

Reply:
comments in this function would be appreciated too